tests/apps/detection/test_box_transform.py

```diff
@@ -59,23 +62,31 @@
         ]
     )
 
+# 2D test data
+boxes_2d = [[0, 1, 2, 2], [0, 0, 1, 1]]
+labels_2d = [1, 0]
+image_size_2d = [1, 2, 2]
+image_2d = np.zeros(image_size_2d)
+
+# Use dict_product for TESTS_2D
 TESTS_2D = []
-boxes = [[0, 1, 2, 2], [0, 0, 1, 1]]
-labels = [1, 0]
-image_size = [1, 2, 2]
-image = np.zeros(image_size)
-for p in TEST_NDARRAYS:
+for params in dict_product(ndarray_type=TEST_NDARRAYS):
+    p = params["ndarray_type"]
     TESTS_2D.append(
-        [{"boxes": p(boxes), "image": p(image), "labels": p(labels)}, p([[[0, 2], [0, 2]], [[1, 0], [0, 0]]])]
+        [{"boxes": p(boxes_2d), "image": p(image_2d), "labels": p(labels_2d)}, p([[[0, 2], [0, 2]], [[1, 0], [0, 0]]])]
     )
 
+# Use dict_product for TESTS_2D_mask
 TESTS_2D_mask = []
-boxes_mask = [[[-1, 0], [0, -1]]]
-for p in TEST_NDARRAYS:
-    TESTS_2D_mask.append([p(boxes_mask), (p([[0.0, 0.0, 2.0, 2.0]]), p([0]))])
-boxes_mask = [[[-1, 0], [0, -1]], [[-1, 1], [1, -1]]]
-for p in TEST_NDARRAYS:
-    TESTS_2D_mask.append([p(boxes_mask), (p([[0.0, 0.0, 2.0, 2.0], [0.0, 0.0, 2.0, 2.0]]), p([0, 1]))])
+boxes_mask_1 = [[[-1, 0], [0, -1]]]
+for params in dict_product(ndarray_type=TEST_NDARRAYS):
+    p = params["ndarray_type"]
+    TESTS_2D_mask.append([p(boxes_mask_1), (p([[0.0, 0.0, 2.0, 2.0]]), p([0]))])
+
+boxes_mask_2 = [[[-1, 0], [0, -1]], [[-1, 1], [1, -1]]]
+for params in dict_product(ndarray_type=TEST_NDARRAYS):
+    p = params["ndarray_type"]
+    TESTS_2D_mask.append([p(boxes_mask_2), (p([[0.0, 0.0, 2.0, 2.0], [0.0, 0.0, 2.0, 2.0]]), p([0, 1]))])
```

Comment:
If there's only one key passed to `dict_product`, this isn't saving any space or complexity so I wouldn't use this here.